### PR TITLE
Improve Prompt Engineering - Real World Examples - Content & Links

### PR DESCRIPTION
**Pull request by iPoetDev**
- intent: Update content & links from May 2023 to Oct 2024

- data: src/data/roadmaps/prompt-engineering/content/
  - 📂 103-real-world

- Link Strategies:
  - Direct links to direct topical pages from each node:
  - Range from beginner to a few intermediate topics for progression/varying audience.
  
- Link / Content authorities, some examples:
  - LearnPrompting.org
  - PromptingGuide.ai
  - Medium.com
  - Ikangai.com
  - Wharton Business (UPenn)
  - Pluralsight
  - IEEE.org
  - TED.com on YouTube
  - IBM Technology on YouTube

- One video “Large language models LLMs for education” could be a generated podcast, so quality/provenance is open for review. However, despite this: it provides a good summation of LLM in education, so it has some utility.

### CODE REVIEW

**Comment by iPoetDev** [comment]
- The content paragraphs conform to the contrib. guidelines, just more contextually generalised and aware.

A few other edits to cover in bullets for newer use cases that were not active in May 2023.

File: src/data/roadmaps/prompt-engineering/content/103-real-world/101-inferring.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Inferring
 
-Making use of LLMs to infer information from text is another common use case. Imagine you are running a famous e-commerce store with thousands of reviews. You want to group the reviews by the sentiment of the review. You can use LLMs to infer the sentiment of the review. 
+Making use of LLMs to infer information from text, i.e. text analysis, is another common use case. A common use case is sentiment analysis, such as grouping e-commerce reviews by sentiment.Beyond this, LLMs have diverse real-world applications, including customer service automation, market research, fraud detection, compliance monitoring, speech-to-text transcription, and anomaly detection. These applications demonstrate the versatility of LLMs in extracting valuable insights from text data, automating processes, and enhancing decision-making across different sectors. By leveraging LLMs, businesses can improve operational efficiency, gain deeper customer insights, and tackle complex challenges in areas such as cybersecurity and regulatory compliance
```
Comment (dansholds) [inline src/data/roadmaps/prompt-engineering/content/103-real-world/101-inferring.md:3]:
```suggestion
Making use of LLMs to infer information from text, i.e. text analysis, is another common use case. A common use case is sentiment analysis, such as grouping e-commerce reviews by sentiment. Beyond this, LLMs have diverse real-world applications, including customer service automation, market research, fraud detection, compliance monitoring, speech-to-text transcription, and anomaly detection. These applications demonstrate the versatility of LLMs in extracting valuable insights from text data, automating processes, and enhancing decision-making across different sectors. By leveraging LLMs, businesses can improve operational efficiency, gain deeper customer insights, and tackle complex challenges in areas such as cybersecurity and regulatory compliance
```

File: src/data/roadmaps/prompt-engineering/content/103-real-world/105-designing-chatbots.md [hunk 2]
```diff
@@ -1,4 +1,9 @@
 # Designing Chatbots
 
-Building chatbots to offer customer support, sales, or other services is a hot topic in the tech industry. LLMs make it possible to build chatbots that can respond to a wide variety of user inputs, and can be trained to respond to new inputs with minimal effort.
+Designing chatbots and other interfaces for real-world usage cases remains a key focus in the tech industry. Large language models (LLMs) continue to enhance chatbot capabilities, allowing them to respond effectively to diverse user inputs with minimal effort and training. The use cases for new user interfaces is expanding beyond traditional chatbots like: voice interfaces, gesture-based interfaces , multimodal AI systems using various input types,  AI agents performing complex tasks autonomously, extended reality (XR) combining AI with AR/VR for immersive experiences. These advancements highlight the evolving nature of user interfaces, driven by technological innovations that promise more engaging and efficient interactions.
```
Comment (dansholds) [inline src/data/roadmaps/prompt-engineering/content/103-real-world/105-designing-chatbots.md:3]:
```suggestion
Designing chatbots and other interfaces for real-world usage cases remains a key focus in the tech industry. Large language models (LLMs) continue to enhance chatbot capabilities, allowing them to respond effectively to diverse user inputs with minimal effort and training. The use cases for new user interfaces is expanding beyond traditional chatbots like: voice interfaces, gesture-based interfaces, multimodal AI systems using various input types,  AI agents performing complex tasks autonomously, extended reality (XR) combining AI with AR/VR for immersive experiences. These advancements highlight the evolving nature of user interfaces, driven by technological innovations that promise more engaging and efficient interactions.
```

**Comment by dansholds** [comment]
LGTM 🚀

**Pull request merged**